src/pubsub/moderation.rs

```diff
@@ -589,6 +593,25 @@ mod tests {
         ));
     }
 
+    #[test]
+    fn vip_added_mod_action() {
+        let source = r#"
+        {
+            "type": "MESSAGE",
+            "data": {
+                "topic": "chat_moderator_actions.691109305.120183018",
+                "message": "{\"type\":\"moderation_action\",\"data\":{\"type\":\"chat_login_moderation\",\"moderation_action\":\"vip\",\"args\":[\"Floikka\"],\"created_by\":\"nam______________________\",\"created_by_user_id\":\"120183018\",\"created_at\":\"2022-12-20T16:41:26.168122804Z\",\"msg_id\":\"\",\"target_user_id\":\"85262774\",\"target_user_login\":\"\",\"from_automod\":false}}"
+            }
+        }"#;
+        let actual = dbg!(Response::parse(source).unwrap());
+        assert!(matches!(
+            actual,
+            Response::Message {
+                data: TopicData::ChatModeratorActions { .. },
```

Comment:
This could be nicer, once [`box_patterns`](https://doc.rust-lang.org/beta/unstable-book/language-features/box-patterns.html) is stable.

Reply:
box patterns are never going to get to stable :(